tests/Service/Results/RecordTest.php

```diff
@@ -18,6 +22,96 @@ public function testAnalyticsData(): void
         $this->assertInstanceOf(AnalyticsData::class, $record->getAnalyticsData());
     }
 
+    /**
+     * @dataProvider provideLinks
+     */
+    public function testGetDecoratedLink(?AnalyticsData $analyticsData, string $link, string $expectedLink): void
+    {
+        if (InstalledVersions::satisfies(new VersionParser(), 'silverstripe/framework', '<5.2.20') && strpos(
+            $link,
+            'www'
+        ) !== false) {
+            // skip external link tests on older versions as the behaviour has changed
+            return;
+        }
```

Comment:
@blueo The presence of `www` doesn't necessarily mean it's external. We have had the odd site in the past that has requested we use fully qualified links for internal link as well.

What was the behaviour change from 5.2?

Reply:
yeah it was only to target our test cases. The new behaviour in 5.2.22 is that external links are excluded from the trailing slash change. If you run this on an earlier version the external links get a slash added. This check is only for the tests as I assumed we didn't want too much effort into older versions - as this is essentially testing the join_links function rather than our params being added